aptos-move/framework/aptos-framework/sources/TestCoin.move

```diff
@@ -47,44 +30,34 @@ module AptosFramework::TestCoin {
         inner: vector<DelegatedMintCapability>,
     }
 
-    /// Events handles.
-    struct TransferEvents has key {
-        sent_events: EventHandle<SentEvent>,
-        received_events: EventHandle<ReceivedEvent>,
-    }
+    public fun initialize(core_framework: &signer, core_resource: &signer): BurnCapability<TestCoin> {
+        SystemAddresses::assert_core_resource(core_resource);
 
-    struct SentEvent has drop, store {
-        amount: u64,
-        to: address,
-    }
+        let (mint_cap, burn_cap) = Coin::initialize_with_capabilities<TestCoin>(
+            core_framework,
+            ASCII::string(b"Test Coin"),
+            ASCII::string(b"TC"),
+            18, /* decimals */
```

Comment:
if we do this many, we're going to think through u64 vs u128 and the perf implications of that being our default coin amount. I don't see there being any need for that many decimals... I think 6, maybe 12 is more than enough.

Reply:
For `u64` numbers max value is `18446744073709551615`, so it's only `18446744.073` max supply for coin with 12 decimals.
Yet it's good question how much decimals should be enough

Reply:
18 is the standard decimals for most tokens on other chains. This originated from 1 ETH = 10**18 wei. u64 is definitely a bit limiting but it also depends on how much gas costs/gas schedule. On EVM chains, gas is measured in 10**9 (gwei or nAVAX). I can update this to be 6 for now but we need a broader discussion on decimals, gas costs, introducing u256.